### ISSUE TITLE
Clarify whether custom matcher message can be DOM node

### ISSUE THREAD

**Issue by cburgmer** [jasmine/jasmine#791]
In http://jasmine.github.io/2.1/custom_matcher.html#section-Failure_Messages there's no mention whether a custom matcher might return a DOM node inside `message`. This has always been possible in the past and is still today (https://github.com/jasmine/jasmine/blob/master/lib/jasmine-core/jasmine-html.js#L290). For example [imagediff](https://www.npmjs.com/package/imagediff) builds on top of this feature.

Please clarify on whether this is still wanted. In this case I'll raise an issue with karma-jasmine to support non-string values inside `message`.

Please note that for CLI runners with a browser in the back the console output will not be able to render the DOM node.

**Comment by slackersoft** [comment]
Jasmine is definitely only expecting to see strings in the failure message as DOM nodes don't make any sense in a node context or when printing the output to a console. That said, I don't think we want to try to maintain the code to enforce that only strings are allowed either.

I think this might be a situation where you just have to know what you and the matcher library you're using are doing to know which makes sense. I wouldn't fault karma for not handling DOM nodes if they're expecting to mostly print to a terminal where that wouldn't make sense.

Basically, a custom matcher is assumed to be specific to the context of the project it is being run in, and thus can be whatever the project thinks is acceptable given how the specs are being run.

**Comment by cburgmer** [comment]
Thanks for the answer.

I understand that adding the complexity of supporting DOM nodes might not be in jasmine's interest. However the current code of the HTML reporter does explicitly support DOM nodes and I would hope we could at least keep this that way as a lot of people might have made their assumptions based on that.

I agree that the customer matcher could handle the complexity of whether the reporter does or does not support anything apart from strings inside messages.

**Comment by slackersoft** [comment]
It sounds like you've got your answer, so I'm going to close this issue for now.

Thanks for using jasmine!
